HabitTracker/Data/DbConnection.cs

```diff
@@ -0,0 +1,172 @@
+﻿using Microsoft.Data.Sqlite;
+
+namespace HabitTracker.Data
+{
+    public class DbConnection
+    {
+        private static SqliteConnection _connection;
+
+        private DbConnection() { }
+        enum Frequency
+        {
+            Daily,
+            Weekly,
+            Monthly
+        }
+
+        public static SqliteConnection GetConnection(bool isStart = false)
+        {
+            if (_connection == null)
+            {
+                const string dbPath = "testDb.db";
+                _connection = new SqliteConnection($"Data Source={dbPath}");
+                _connection.Open();
+                Console.WriteLine("The database has been connected.");
+               
+                if (isStart)
+                {
+                    DbConnection.CreateTable("Habits");
+                    DbConnection.CreateTable("Records");
+                    DbConnection.SeedData();
+                }
+
+
+                _connection.Close();
+            }
+            return _connection;
+        }
+
+        public static void CloseConnection()
+        {
+            if (_connection != null)
+            {
+                _connection.Close();
+                _connection.Dispose();
+                _connection = null;
+                Console.WriteLine("The database connection has been closed.");
+            }
+        }
+
+        public static void CreateTable(string name)
+        {
+            try
+            {
+                var command = _connection.CreateCommand();
+                if (name == "Habits")
+                {
+                    command.CommandText =
+                    @"
+                        CREATE TABLE IF NOT EXISTS Habits (
+                            Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                            Name TEXT NOT NULL,                    
+                            Frequency TEXT NOT NULL,                    
+                            TimesPerPeriod INTEGER NOT NULL,                    
+                            StartDate TEXT NOT NULL                    
+                        )
+                    ";
+                    command.ExecuteNonQuery();
+                }
+                else
+                {
+                    command.CommandText =
+                    @"
+                        CREATE TABLE IF NOT EXISTS Records (
+                            Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                            Name TEXT NOT NULL,                    
+                            HabitDate TEXT NOT NULL,
+                            HabitId INTEGER,
+                            FOREIGN KEY (HabitId) REFERENCES Habits(Id) ON DELETE CASCADE
+                        )
+                    ";
+                    command.ExecuteNonQuery();
+                }
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Error creating table: " + ex.Message);
```

Comment:
🟠 **String Interpolation**

💡 Use string interpolation where possible instead of string concatenation.

🔧 `Console.WriteLine($"Error creating table: {ex.Message}");`